### PR TITLE
feat: rewrite e2e test(upstream_chash_hash_on_test) with ginkgo

### PR DESCRIPTION
**Pull request by Jaycean**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
#1500

___
### New feature or improvement
-  Rewrite e2e test(upstream_chash_hash_on_test) with ginkgo

### CODE REVIEW

**Comment by nic-chen** [comment]
@Jaycean  is this pr ready ?

**Comment by Jaycean** [comment]
> @Jaycean is this pr ready ?

Yes, but I'm going to check it again before opening it.

**Comment by Jaycean** [comment]
cc @juzhiyuan @liuxiran FE e2e test failed. PTAL.

**Comment by juzhiyuan** [comment]
ping @guoqqqi to take a look

**Review by nic-chen — APPROVED** [review]

File: api/test/e2enew/upstream/upstream_chash_hash_on_test.go [hunk 1]
```diff
@@ -0,0 +1,585 @@
...
+		time.Sleep(time.Duration(500) * time.Millisecond)
+		basepath := base.APISIXHost
+		res := map[string]int{}
+		for i := 0; i <= 3; i++ {
+			url := basepath + "/server_port?var=2&var2=" + strconv.Itoa(i)
+			req, err := http.NewRequest("GET", url, nil)
+			assert.Nil(t, err)
+			req.Header.Add("custom_header", `custom-one`)
+			resp, err := http.DefaultClient.Do(req)
+			assert.Nil(t, err)
```
Comment (starsz) [inline api/test/e2enew/upstream/upstream_chash_hash_on_test.go:89]: You can call the resp.Body.Close here.

File: api/test/e2enew/upstream/upstream_chash_hash_on_test.go [hunk 2]
```diff
@@ -0,0 +1,585 @@
...
+			respBody, err := ioutil.ReadAll(resp.Body)
+			assert.Nil(t, err)
+			body := string(respBody)
+			if _, ok := res[body]; !ok {
+				res[body] = 1
+			} else {
+				res[body]++
+			}
+			resp.Body.Close()
```
Comment (starsz) [inline api/test/e2enew/upstream/upstream_chash_hash_on_test.go:186]: Ditto.

File: api/test/e2enew/upstream/upstream_chash_hash_on_test.go [hunk 3]
```diff
@@ -0,0 +1,585 @@
...
```
Comment (starsz) [inline api/test/e2enew/upstream/upstream_chash_hash_on_test.go:211]: Ditto.

File: api/test/e2enew/upstream/upstream_chash_hash_on_test.go [hunk 4]
```diff
@@ -0,0 +1,585 @@
...
+			resp, err := http.DefaultClient.Do(req)
+			assert.Nil(t, err)
+			respBody, err := ioutil.ReadAll(resp.Body)
+			body := string(respBody)
+			if _, ok := res[body]; !ok {
+				res[body] = 1
+			} else {
+				res[body]++
+			}
+			resp.Body.Close()
```
Comment (starsz) [inline api/test/e2enew/upstream/upstream_chash_hash_on_test.go:553]: Ditto.

File: api/test/e2enew/upstream/upstream_chash_hash_on_test.go [hunk 5]
```diff
@@ -0,0 +1,585 @@
...
+			assert.Nil(t, err)
+			respBody, err := ioutil.ReadAll(resp.Body)
+			assert.Nil(t, err)
+			body := string(respBody)
+			if _, ok := res[body]; !ok {
+				res[body] = 1
+			} else {
+				res[body]++
+			}
+			resp.Body.Close()
```
Comment (starsz) [inline api/test/e2enew/upstream/upstream_chash_hash_on_test.go:405]: Ditto.

File: api/test/e2enew/upstream/upstream_chash_hash_on_test.go [hunk 6]
```diff
@@ -0,0 +1,585 @@
...
+			resp, err := http.DefaultClient.Do(req)
+			assert.Nil(t, err)
+			respBody, err := ioutil.ReadAll(resp.Body)
+			body := string(respBody)
+			if _, ok := res[body]; !ok {
+				res[body] = 1
+			} else {
+				res[body]++
+			}
+			resp.Body.Close()
```
Comment (starsz) [inline api/test/e2enew/upstream/upstream_chash_hash_on_test.go:297]: Ditto.

File: api/test/e2enew/upstream/upstream_chash_hash_on_test.go (hunk 1 above)
Comment (Jaycean) [reply api/test/e2enew/upstream/upstream_chash_hash_on_test.go:89]: done.

File: api/test/e2enew/upstream/upstream_chash_hash_on_test.go (hunk 3 above)
Comment (Jaycean) [reply api/test/e2enew/upstream/upstream_chash_hash_on_test.go]: done.

File: api/test/e2enew/upstream/upstream_chash_hash_on_test.go (hunk 4 above)
Comment (Jaycean) [reply api/test/e2enew/upstream/upstream_chash_hash_on_test.go]: done.

File: api/test/e2enew/upstream/upstream_chash_hash_on_test.go (hunk 5 above)
Comment (Jaycean) [reply api/test/e2enew/upstream/upstream_chash_hash_on_test.go]: done.

File: api/test/e2enew/upstream/upstream_chash_hash_on_test.go (hunk 6 above)
Comment (Jaycean) [reply api/test/e2enew/upstream/upstream_chash_hash_on_test.go]: done.

File: api/test/e2enew/upstream/upstream_chash_hash_on_test.go (hunk 2 above)
Comment (Jaycean) [reply api/test/e2enew/upstream/upstream_chash_hash_on_test.go]: done.

**Comment by Jaycean** [comment]
cc @starsz PTAL, Thks.

**Review by starsz — APPROVED** [review]

**Pull request merged**